### ISSUE TITLE
easy undocking and window focus

### ISSUE THREAD

**Issue by grotesque** [freeplane/freeplane#1433]
When I want use freeplane on multiple monitors I can not do it without annoyances.

1. There is no undock current mm shortcut
2. If I alt+tab or select the main freeplane window or if I alt+tab or select an undocked window both (the undocked and main freeplane window) of them don't get focus/maximized/visible. I have to open freeplane undocked window and freeplane main window separately.

**Comment by euu2021** [comment]
> 1. There is no undock current mm shortcut

Sometime ago, I tried to include it in a script by using the internal methods (from [Infonode](https://github.com/ebourg/infonode/blob/672cc6b8323525bc5c1caad0c4c39e8ea3219828/src/net/infonode/docking/FloatingWindow.java)), but I failed. The thing almost worked, but, in the end, it failed. Maybe, someone more skilled than me can create that script.

> If I alt+tab or select the main freeplane window or if I alt+tab or select an undocked window both (the undocked and main freeplane window) of them don't get focus/maximized/visible. I have to open freeplane undocked window and freeplane main window separately.

I see that the behavior that you describe can be useful in some situations. But, I must say that the current behavior is the most useful for me in most situations.

**Comment by euu2021** [comment]
> If I alt+tab or select the main freeplane window or if I alt+tab or select an undocked window both (the undocked and main freeplane window) of them don't get focus/maximized/visible. I have to open freeplane undocked window and freeplane main window separately.

Are you using Windows? I think it can be solved with an AutoHotkey script.

**Comment by grotesque** [comment]
> I think it can be solved with an AutoHotkey script.

Yes I'm using windows. Can you elabaorate a bit about the script you're talking about?

**Comment by euu2021** [comment]
> Yes I'm using windows. Can you elabaorate a bit about the script you're talking about?

Try something like this, for example: https://www.autohotkey.com/boards/viewtopic.php?t=31843

And it's possible to make a script that makes all the windows come to the foreground when any of them is selected, automatically. Ask for help in the autohotkey forum, and they will help you in no time.

**Comment by grotesque** [comment]
Thanks @euu2021 
Following ahk script brings to front all windows opened by freeplane on pressing win+f

```
#f::
WinGet, id, list, ahk_class SunAwtFrame
Loop, %id%
{
    this_id := id%A_Index%
    WinActivate, ahk_id %this_id%
}

Goto, GEnd

Capslock & F2::Reload

GEnd:
return
```
